go/httpdispatcher/proxy.go

```diff
@@ -527,6 +524,54 @@ func (p *Proxy) detectPortalURL(r *http.Request) (bool, url.URL) {
 	return found, PortalURL
 }
 
+func (p *Proxy) handleDetectionMechanismRegister(ctx context.Context, w http.ResponseWriter, r *http.Request, fqdn string) {
+	var ipAddress string
+	ipAddress = p.getIP(ctx, r)
+
+	if ipAddress != "" {
+		MAC, err := p.IP2Mac(ctx, ipAddress)
+
+		if err == nil {
+			if p.nodeStatus(ctx, MAC) && passThrough.checkDetectionMechanisms(ctx, fqdn) {
+				log.LoggerWContext(ctx).Info("Device register and match the portal detection mechanism for " + MAC)
+				p.reverse(ctx, w, r, r.Host)
+			}
+		}
+	}
+}
+
+// nodeStatus search for status of the device
+func (p *Proxy) nodeStatus(ctx context.Context, mac string) bool {
```

Comment:
Should be called nodeIsReg since you're not returning the status column value but whether or not the status is reg